datastore/api/snippets.py

```diff
@@ -0,0 +1,823 @@
+        'done': False,
+        # Integer
+        'priority': 4,
+        # Float
+        'percent_complete': 10.5,
+    })
+    # [END properties]
+
+    return task
+
+
+def array_value(client):
+    # [START array_value]
+    task = datastore.Entity()
+    task.update({
+        'tags': [
+            'fun',
+            'programming'
+        ],
+        'collaborators': [
+            'alice',
+            'bob'
+        ]
+    })
+    # [END array_value]
+
+    return task
+
+
+def upsert(client):
+    # [START upsert]
+    complete_key = client.key('Task', 'sample_task')
+
+    task = datastore.Entity(key=complete_key)
+
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+
+    client.put(task)
+    # [END upsert]
+
+    return task
+
+
+def insert(client):
+    # [START insert]
+    incomplete_key = client.key('Task')
+
+    task = datastore.Entity(key=incomplete_key)
+
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+
+    client.put(task)
+    # [END insert]
+
+    return task
+
+
+def update(client):
+    # Create the entity we're going to update.
+    upsert(client)
+
+    # [START update]
+    key = client.key('Task', 'sample_task')
+    task = client.get(key)
+
+    task['done'] = True
+
+    client.put(task)
+    # [END update]
+
+    return task
+
+
+def lookup(client):
+    # Create the entity that we're going to look up.
+    upsert(client)
+
+    # [START lookup]
+    key = client.key('Task', 'sample_task')
+    task = client.get(key)
+    # <Entity
+    #   [{'kind': 'Task', 'name': 'sample_task'}]
+    #   {
+    #       'priority': 4,
+    #       'type': 'Personal',
+    #       'done': False,
+    #       'description': 'Learn Cloud Datastore'
+    # }>
+    # [END lookup]
+
+    return task
+
+
+def delete(client):
+    # Create the entity we're going to delete.
+    upsert(client)
+
+    # [START delete]
+    key = client.key('Task', 'sample_task')
+    client.delete(key)
+    # [END delete]
+
+    return key
+
+
+def batch_upsert(client):
+    # [START batch_upsert]
+    task1 = datastore.Entity(client.key('Task', 1))
+
+    task1.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+
+    task2 = datastore.Entity(client.key('Task', 2))
+
+    task2.update({
+        'type': 'Work',
+        'done': False,
+        'priority': 8,
+        'description': 'Integrate Cloud Datastore'
+    })
+
+    client.put_multi([task1, task2])
+    # [END batch_upsert]
+
+    return task1, task2
+
+
+def batch_lookup(client):
+    # Create the entities we will lookup.
+    batch_upsert(client)
+
+    # [START batch_lookup]
+    keys = [
+        client.key('Task', 1),
+        client.key('Task', 2)
+    ]
+
+    tasks = client.get_multi(keys)
```

Comment:
Again, we need to be consistent with the other APIs so that the docs can also have a consistent feel. I think this was one of the problems with the existing docs -- each language had slightly different code so if the docs made references to code pieces they would be incorrect in certain languages.